### PR TITLE
bugfix: fix the incomplete delete operation of runtime-rs

### PR DESCRIPTION
**Pull request by YushuoEdge**
This PR consists of three commits:

- The first commit solves the problem of resource deletion of Containers that are created but never started.
- The second commit solves the problem of checking states in start_container by mistake.
- The third commit solves the problem of inconsistent information stored after the cleanup of container's rootfs and volumes.

Fixes: #6743

Signed-off-by: Yushuo <y-shuo@linux.alibaba.com>

### CODE REVIEW

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 1]
```diff
@@ -305,10 +306,70 @@ impl Container {
...
+
+        // ignore the stop when container status has already been stopped.
+        let status = inner.init_process.get_status().await;
+        if status == ProcessStatus::Stopped {
+            return Ok(());
+        }
+
+        match container_process.process_type {
+            ProcessType::Container => {
+                inner
```
Comment (Tim-0731-Hzt) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:318]: As both `Exec` and `Container`  `ProcessType` will do `stop_process,` I think you can just do it once, and use with_context here for different process type

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 2]
```diff
@@ -305,10 +306,70 @@ impl Container {
 
     pub async fn stop_process(&self, container_process: &ContainerProcess) -> Result<()> {
         let mut inner = self.inner.write().await;
-        inner
-            .stop_process(container_process, true)
+
+        // ignore the stop when container status has already been stopped.
+        let status = inner.init_process.get_status().await;
+        if status == ProcessStatus::Stopped {
```
[the remark this answers is not in the document]
Comment (YushuoEdge) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:312]: This function will be called from wait_process and delete_process. If the user calls `kill` and `delete`, then this function will be called twice. We need to ignore the second.

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs (hunk 1 above)
Comment (YushuoEdge) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs]: fixed

File: src/runtime-rs/crates/resource/src/rootfs/mod.rs [hunk 3]
```diff
@@ -139,3 +155,8 @@ impl RootFsResource {
 fn is_single_layer_rootfs(rootfs_mounts: &[Mount]) -> bool {
     rootfs_mounts.len() == 1
 }
+
+pub fn generate_rootfs_id() -> String {
```
Comment (Tim-0731-Hzt) [inline src/runtime-rs/crates/resource/src/rootfs/mod.rs:159]: pub(crate)

Comment (YushuoEdge) [reply src/runtime-rs/crates/resource/src/rootfs/mod.rs]: Fixed. Thanks!

File: src/runtime-rs/crates/resource/src/volume/mod.rs [hunk 4]
```diff
@@ -116,3 +130,8 @@ fn is_skip_volume(_m: &oci::Mount) -> bool {
     // TODO: support volume check
     false
 }
+
+pub fn generate_volume_id() -> String {
```
[the remark this answers is not in the document]
Comment (YushuoEdge) [reply src/runtime-rs/crates/resource/src/volume/mod.rs]: Fixes. Thanks!

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 5]
```diff
@@ -305,10 +306,65 @@ impl Container {
...
+                    Ok(agent::Empty::new())
+                } else {
+                    Err(e)
+                }
+            })?;
+
+        let mut inner = self.inner.write().await;
+
+        // cleanup rootfs and volumes, and notify resource manager to keep states persistent
+        let cleaned_rootfs = inner.clean_rootfs().await?;
```
Comment (Tim-0731-Hzt) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:360]: We could consider doing this with the `resource manager`. As for block volume and block rootfs cleanup, we need to use the `device manager `to do this. Then, we need to pass the `device manager` as the parameter for the cleanup function. But in the current implementation, it's unable to fetch `device manager`. So, my suggestion is to pass `Vec<Arc<dyn Rootfs>>` and `Vec<Arc<dyn Volume>>` to `resource manager`.

Comment (YushuoEdge) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs]: Update.
Now the cleanup operation is doing in `ResourceManager`, and also the unremoved rootfs and volumes will be returned to `Container`. WDYT?

**Pull request closed without merging**